assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js

```diff
@@ -99,10 +100,46 @@ export default function UserDimensionsPieChart( {
 			}
 		};
 
+		const closeToolTip = () =>
+			setValues( {
+				[ UI_DIMENSION_VALUE ]: '',
+				[ UI_DIMENSION_COLOR ]: '',
+				[ UI_ACTIVE_ROW_INDEX ]: null,
+			} );
+
+		// When the user hits the 'escape' key and the tooltip is open, close the tooltip.
+		const onEscape = ( event = {} ) => {
+			if (
+				event?.keyCode === ESCAPE &&
+				!! currentContainerRef.querySelector(
+					'.google-visualization-tooltip'
+				)
+			) {
+				closeToolTip();
+			}
+		};
+
+		// When the use clicks outside of the chart and the tooltip is open, close the tooltip.
+		const onExitClick = ( event = {} ) => {
+			if (
+				!! currentContainerRef.querySelector(
+					'.google-visualization-tooltip'
+				) &&
+				! event?.target?.closest(
+					'.googlesitekit-widget--analyticsAllTraffic__dimensions-chart'
+				)
+			) {
+				closeToolTip();
+			}
+		};
```

Comment:
@johnPhillips we also need to define these callbacks after the `currentContainerRef` constant is defined. I know that the variable will be hoisted, but having it defined before callbacks helps to read the code.

```js
const currentContainerRef = containerRef.current;

const closeToolTip = () => ...

// When the user hits the 'escape' key and the tooltip is open, close the tooltip.
const onEscape = ( event = {} ) => ...

// When the use clicks outside of the chart and the tooltip is open, close the tooltip.
const onExitClick = ( event = {} ) => ...
```

Reply:
Sure thing, I'll do that now.